components/squad/squad.lua

```diff
@@ -0,0 +1,76 @@
+local Class = require('Module:Class')
+local Arguments = require('Module:Arguments')
+local String = require('Module:String')
+
+local Squad = Class.new()
+
+function Squad:init(frame)
+	self.frame = frame
+	self.args = Arguments.getArgs(frame)
+	self.root = mw.html.create('div')
+	self.root	:addClass('table-responsive')
+				-- TODO: is this needed?
+				:css('margin-bottom', '10px')
+				-- TODO: is this needed?
+				:css('padding-bottom', '0px')
+
+	self.content = mw.html.create('table')
+	self.content:addClass('wikitable wikitable-striped roster-card')
+
+	return self
+end
+
+function Squad:title()
+	local titleText = String.isEmpty(self.args.title) and 'Active Squad' or self.args.title
+
+	local titleContainer = mw.html.create('tr')
+
+	local titleRow = mw.html.create('th')
+	titleRow:addClass('large-only')
+			:attr('colspan', '1')
+			:wikitext(titleText)
+
+	local titleRow2 = mw.html.create('th')
+	titleRow2:addClass('large-only')
+			:attr('colspan', '10')
+			:css('border-bottom', '1px solid #bbbbbb')
+			:wikitext(titleText)
+
+	titleContainer:node(titleRow):node(titleRow2)
+	self.content:node(titleContainer)
+
+	return self
+end
+
+function Squad:header()
+	local makeHeader = function(wikiText)
+		local headerCell = mw.html.create('th')
+
+		if wikiText == nil then
+			return headerCell
+		end
+
+		return headerCell:wikitext(wikiText):addClass('divCell')
+	end
+
+	local headerRow = mw.html.create('tr'):addClass('HeaderRow')
+	headerRow	:node(makeHeader('ID'))
+				:node(makeHeader('Name'))
+				:node(makeHeader())
+				:node(makeHeader('Join Date'))
+	self.content:node(headerRow)
```

Comment:
How to add custom headers for a wiki?
How to add Leave Date for Former Squad table?

Reply:
Simply extend the class and override this function